File: lib/src/main/java/fr/o80/sample/lib/ui/MainActivity.java
package fr.o80.sample.lib.ui;

import android.os.Bundle;
import android.os.PersistableBundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;

import javax.inject.Inject;

import fr.o80.sample.lib.core.Feature;
import fr.o80.sample.lib.core.LibApplication;
import fr.o80.sample.lib.core.LibConfiguration;

/**
 * @author Olivier Perez
 */
public class MainActivity extends AppCompatActivity {

    @Inject
    protected LibConfiguration libConfiguration;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        ((LibApplication)getApplication()).component().inject(this);

        Feature home = libConfiguration.getHome();
        home.open(this);

        finish();
    }
}

File: lib/src/main/java/fr/o80/sample/lib/ui/MainActivity.kt
package fr.o80.sample.lib.ui

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import fr.o80.sample.lib.core.LibApplication
import fr.o80.sample.lib.core.LibConfiguration
import javax.inject.Inject

/**
 * @author Olivier Perez
 */
class MainActivity : AppCompatActivity() {

    @Inject
    lateinit var libConfiguration: LibConfiguration

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        (application as LibApplication).component.inject(this)

        with(libConfiguration) {
            home.open(this@MainActivity)
        }

        finish()
    }
}
